File: src/admin_dashboard_canister/src/types/adminRequest.ts
// Types matching the owners_canister Rust types
export interface AdminRequest {
  id: string;
  requester_principal: string;
  name: string;
  reason: string;
  status: 'Pending' | 'Approved' | 'Rejected';
  requested_at: bigint;
  processed_at?: bigint;
  processed_by?: string;
  rejection_reason?: string;
}

export interface CreateAdminRequest {
  name: string;
  reason: string;
}

export interface ProcessAdminRequest {
  request_id: string;
  approve: boolean;
  rejection_reason?: string;
}

export type AdminRequestStatus = 'Pending' | 'Approved' | 'Rejected';

File: src/owners_canister/src/types/admin_request.rs
use candid::{CandidType, Principal};
use serde::{Deserialize, Serialize};

// Admin request status
#[derive(Clone, CandidType, Deserialize, Serialize, PartialEq, Debug)]
pub enum AdminRequestStatus {
    Pending,
    Approved,
    Rejected,
}

// Admin request struct
#[derive(Clone, CandidType, Deserialize, Serialize)]
pub struct AdminRequest {
    pub id: String,
    pub requester_principal: Principal,
    pub name: String,
    pub reason: String,
    pub status: AdminRequestStatus,
    pub requested_at: u64,
    pub processed_at: Option<u64>,
    pub processed_by: Option<Principal>,
    pub rejection_reason: Option<String>,
}

// Input for creating admin request
#[derive(Clone, CandidType, Deserialize, Serialize)]
pub struct CreateAdminRequest {
    pub name: String,
    pub reason: String,
}

// Input for processing admin request
#[derive(Clone, CandidType, Deserialize, Serialize)]
pub struct ProcessAdminRequest {
    pub request_id: String,
    pub approve: bool,
    pub rejection_reason: Option<String>,
}
